### PR TITLE
Bizy 639 pre commit eslint hook

### PR DESCRIPTION
**Pull request by chebotiuk**
I have also moved postinstall sh scripts (that installs scripts to bizy-web directory) to `bizy-scripts`

### CODE REVIEW

File: git-hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+cd "$(git rev-parse --show-toplevel)"
+ESLINT="node_modules/.bin/eslint"
+pwd
+
+if [[ ! -x "$ESLINT" ]]; then
+  printf "\t\033[41mPlease install ESlint\033[0m (npm install eslint)\n"
```
Comment (tarapon) [inline git-hooks/pre-commit]: `yarn add eslint` in comment

File: git-hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  exit 1
+fi
+
+STAGED_FILES=($(git diff --cached --name-only --diff-filter=ACM | grep ".jsx\{0,1\}$"))
+
+echo "ESLint'ing ${#STAGED_FILES[@]} files"
```
Comment (tarapon) [inline git-hooks/pre-commit]: I believe this is noise

File: git-hooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+if [[ "$STAGED_FILES" = "" ]]; then
+  exit 0
+fi
+
+$ESLINT "${STAGED_FILES[@]}" --fix
+
+ESLINT_EXIT="$?"
+
+# Re-add files since they may have been fixed
+git add "${STAGED_FILES[@]}"
```
Comment (tarapon) [inline git-hooks/pre-commit]: not sure if it should automatically fix or just display errors. @helios1138 what do you think?

File: shell/postinstall.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+if [ -d "../../.git" ]
```
Comment (tarapon) [inline shell/postinstall.sh]: it should be called by npm hook, defined in package.json

File: git-hooks/pre-commit (hunk 3 above)
Comment (helios1138) [reply git-hooks/pre-commit]: yeah, I also don't like that the script would fix the errors automatically

File: git-hooks/pre-commit [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+cd "$(git rev-parse --show-toplevel)"
+ESLINT="node_modules/.bin/eslint"
+pwd
+
+if [[ ! -x "$ESLINT" ]]; then
+  printf "\t\033[41mPlease install ESlint\033[0m (npm install eslint)\n"
+  exit 1
+fi
+
+STAGED_FILES=($(git diff --cached --name-only --diff-filter=ACM | grep ".jsx\{0,1\}$"))
```
Comment (tarapon) [inline git-hooks/pre-commit]: git diff HEAD

File: git-hooks/pre-commit [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+$ESLINT "${STAGED_FILES[@]}" --fix
+
+ESLINT_EXIT="$?"
+
+# Re-add files since they may have been fixed
+git add "${STAGED_FILES[@]}"
+
+if [[ "${ESLINT_EXIT}" == 0 ]]; then
+  printf "\n\033[42mCOMMIT SUCCEEDED\033[0m\n"
```
Comment (tarapon) [inline git-hooks/pre-commit]: Remove this printf

**Comment by chebotiuk** [comment]
@tarapon @helios1138 done

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bizy-scripts",
-  "version": "1.0.7",
+  "version": "1.0.8",
```
Comment (helios1138) [inline package.json]: `1.1.0`, if we're following the semver

Comment (chebotiuk) [reply package.json]: @helios1138 oh, sure, agree with you

Comment (chebotiuk) [reply package.json]: @helios1138 done

**Comment by chebotiuk** [comment]
@tarapon up

**Pull request merged**